From 22f54417162ba9a705342543f9dc426fb08bdb61 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Tue, 12 Jun 2012 10:27:51 +0800
Subject: [PATCH 005/602] ENGR00000004 Separate the SoC configures from
 BoardConfig

Some configures are defined to identify the SoC ability.
Like BOARD_SOC_TYPE, HAVE_FSL_IMX_GPU, TARGET_CPU_SMP, etc.
We move them out of the BoardConfig to a separate make
file, this can reuse and share more configures.

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 arm2_6dq/BoardConfig.mk                | 12 +-----------
 arm2_6sl/BoardConfig.mk                | 14 +-------------
 imx6/BoardConfigCommon.mk              | 10 ++--------
 imx6/sabresd/SabreSDBoardConfigComm.mk |  7 -------
 imx6/soc/imx6dq.mk                     | 16 ++++++++++++++++
 imx6/soc/imx6sl.mk                     | 16 ++++++++++++++++
 sabresd_6dq/BoardConfig.mk             |  8 +-------
 7 files changed, 37 insertions(+), 46 deletions(-)
 create mode 100644 imx6/soc/imx6dq.mk
 create mode 100644 imx6/soc/imx6sl.mk

diff --git a/arm2_6dq/BoardConfig.mk b/arm2_6dq/BoardConfig.mk
index be023cc..8f766c6 100755
--- a/arm2_6dq/BoardConfig.mk
+++ b/arm2_6dq/BoardConfig.mk
@@ -2,18 +2,8 @@
 # Product-specific compile-time definitions.
 #
 
+include device/fsl/imx6/soc/imx6dq.mk
 include device/fsl/imx6/arm2/BoardConfigArm2.mk
 
-BOARD_SOC_TYPE := IMX6DQ
-
-USE_OPENGL_RENDERER := true
-TARGET_CPU_SMP := true
-
-HAVE_FSL_IMX_GPU := true
-HAVE_FSL_IMX_IPU := true
-HAVE_FSL_IMX_CODEC := true
-
-BOARD_KERNEL_BASE := 0x10800000
 BOARD_KERNEL_CMDLINE := console=ttymxc0,115200 init=/init androidboot.console=ttymxc0
-TARGET_KERNEL_DEFCONF := imx6_android_defconfig
 TARGET_BOOTLOADER_CONFIG := 6q:mx6q_arm2_android_config 6dl:mx6dl_arm2_config
diff --git a/arm2_6sl/BoardConfig.mk b/arm2_6sl/BoardConfig.mk
index 350456d..e760643 100755
--- a/arm2_6sl/BoardConfig.mk
+++ b/arm2_6sl/BoardConfig.mk
@@ -2,23 +2,11 @@
 # Product-specific compile-time definitions.
 #
 
+include device/fsl/imx6/soc/imx6sl.mk
 include device/fsl/imx6/arm2/BoardConfigArm2.mk
 
-BOARD_SOC_TYPE := IMX6SL
-
-USE_OPENGL_RENDERER := false
-TARGET_CPU_SMP := false
-
-HAVE_FSL_IMX_GPU := false
-HAVE_FSL_IMX_IPU := false
-HAVE_FSL_IMX_CODEC := false
 USE_CAMERA_STUB := true
 BOARD_HAVE_IMX_CAMERA := false
 
-BOARD_KERNEL_BASE := 0x80800000
 BOARD_KERNEL_CMDLINE := console=ttymxc0,115200 init=/init androidboot.console=ttymxc0
-TARGET_KERNEL_DEFCONF := imx6s_android_defconfig
 TARGET_BOOTLOADER_CONFIG := mx6sl:mx6sl_arm2_android_config
-
-TARGET_HAVE_IMX_GRALLOC := false
-TARGET_HAVE_IMX_HWCOMPOSER := false
diff --git a/imx6/BoardConfigCommon.mk b/imx6/BoardConfigCommon.mk
index 7121547..6bca021 100644
--- a/imx6/BoardConfigCommon.mk
+++ b/imx6/BoardConfigCommon.mk
@@ -11,10 +11,11 @@ TARGET_NO_BOOTLOADER := true
 TARGET_NO_KERNEL := false
 TARGET_NO_RECOVERY := false
 TARGET_NO_RADIOIMAGE := true
-TARGET_KERNEL_DEFCONF := imx6_android_defconfig
 
 TARGET_PROVIDES_INIT_RC := true
 
+BOARD_SOC_CLASS := IMX6
+
 #BOARD_USES_GENERIC_AUDIO := true
 BOARD_USES_ALSA_AUDIO := true
 BUILD_WITH_ALSA_UTILS := true
@@ -30,9 +31,6 @@ BOARD_MODEM_ID := EM750M
 BOARD_MODEM_HAVE_DATA_DEVICE := true
 BOARD_HAVE_IMX_CAMERA := true
 
-HAVE_FSL_IMX_CODEC := true
-TARGET_HAVE_IMX_GRALLOC := true
-TARGET_HAVE_IMX_HWCOMPOSER = true
 BUILD_WITHOUT_FSL_DIRECTRENDER := false
 BUILD_WITHOUT_FSL_XEC := true
 
@@ -43,8 +41,6 @@ EXCLUDED_CODEC_V2_BUILD := true
 
 BUILD_WITH_GST := false
 
-BUILD_WITH_GST := false
-
 # for ums config, only export one partion instead of the whole disk
 UMS_ONEPARTITION_PER_DISK := true
 
@@ -60,5 +56,3 @@ BOARD_RECOVERY_PARTITION_SIZE = 10M
 
 BOARD_SYSTEMIMAGE_PARTITION_SIZE := 209715200
 BOARD_FLASH_BLOCK_SIZE := 4096
-
-INTERNAL_USERIMAGES_USE_UBIFS=true
diff --git a/imx6/sabresd/SabreSDBoardConfigComm.mk b/imx6/sabresd/SabreSDBoardConfigComm.mk
index 32b36e9..e4504d7 100755
--- a/imx6/sabresd/SabreSDBoardConfigComm.mk
+++ b/imx6/sabresd/SabreSDBoardConfigComm.mk
@@ -6,7 +6,6 @@
 include device/fsl/imx6/BoardConfigCommon.mk
 
 TARGET_BOOTLOADER_BOARD_NAME := SABRELITE
-BOARD_SOC_CLASS := IMX6
 
 # Wifi
 BOARD_WLAN_DEVICE := ar6003
@@ -28,10 +27,6 @@ WIFI_COMPAT_MODULE_NAME  := "compat"
 WIFI_COMPAT_MODULE_ARG   := ""
 WIFI_TEST_INTERFACE      := "sta"
 
-
-BOARD_HAVE_VPU := true
-HAVE_FSL_IMX_GPU := true
-HAVE_FSL_IMX_IPU := true
 BOARD_MODEM_VENDOR := AMAZON
 
 BOARD_HAVE_HARDWARE_GPS := true
@@ -49,6 +44,4 @@ TARGET_USERIMAGES_USE_EXT4 := true
 # atheros 3k BT
 BOARD_USE_AR3K_BLUETOOTH := true
 
-BOARD_KERNEL_BASE := 0x10800000
-
 USE_ION_ALLOCATOR := true
diff --git a/imx6/soc/imx6dq.mk b/imx6/soc/imx6dq.mk
new file mode 100644
index 0000000..c6268ce
--- /dev/null
+++ b/imx6/soc/imx6dq.mk
@@ -0,0 +1,16 @@
+#
+# SoC-specific compile-time definitions.
+#
+
+BOARD_SOC_TYPE := IMX6DQ
+BOARD_HAVE_VPU := true
+HAVE_FSL_IMX_GPU := true
+HAVE_FSL_IMX_IPU := true
+BOARD_KERNEL_BASE := 0x10800000
+TARGET_KERNEL_DEFCONF := imx6_android_defconfig
+HAVE_FSL_IMX_CODEC := true
+TARGET_HAVE_IMX_GRALLOC := true
+TARGET_HAVE_IMX_HWCOMPOSER = true
+USE_OPENGL_RENDERER := true
+TARGET_CPU_SMP := true
+
diff --git a/imx6/soc/imx6sl.mk b/imx6/soc/imx6sl.mk
new file mode 100644
index 0000000..8acf8e8
--- /dev/null
+++ b/imx6/soc/imx6sl.mk
@@ -0,0 +1,16 @@
+#
+# SoC-specific compile-time definitions.
+#
+
+BOARD_SOC_TYPE := IMX6SL
+BOARD_HAVE_VPU := true
+HAVE_FSL_IMX_GPU := false
+HAVE_FSL_IMX_IPU := false
+BOARD_KERNEL_BASE := 0x80800000
+TARGET_KERNEL_DEFCONF := imx6s_android_defconfig
+HAVE_FSL_IMX_CODEC := false
+TARGET_HAVE_IMX_GRALLOC := false
+TARGET_HAVE_IMX_HWCOMPOSER = false
+USE_OPENGL_RENDERER := false
+TARGET_CPU_SMP := false
+
diff --git a/sabresd_6dq/BoardConfig.mk b/sabresd_6dq/BoardConfig.mk
index 8b86603..4dea943 100755
--- a/sabresd_6dq/BoardConfig.mk
+++ b/sabresd_6dq/BoardConfig.mk
@@ -2,14 +2,8 @@
 # Product-specific compile-time definitions.
 #
 
+include device/fsl/imx6/soc/imx6dq.mk
 include device/fsl/imx6/sabresd/SabreSDBoardConfigComm.mk
 
-BOARD_SOC_TYPE := IMX6DQ
-
-USE_OPENGL_RENDERER := true
-TARGET_CPU_SMP := true
-
 BOARD_KERNEL_CMDLINE := console=ttymxc0,115200 init=/init video=mxcfb0 fbmem=10M vmalloc=400M androidboot.console=ttymxc0
-
 TARGET_BOOTLOADER_CONFIG := 6q:mx6q_sabresd_android_config 6dl:mx6dl_sabresd_android_config
-
-- 
1.8.0

